### PR TITLE
Feature Extraction - Refactor command-line script into a Python module

### PR DESCRIPTION
**Pull request by JooZef315**
### Purpose
This PR converts an existing command-line script `launch_feature_extraction.py` into a Python module for better structure, reuse, and test coverage.

### Summary of Changes
1. Refactored the original script into a Python module (`feature_extraction_module.py`).
3. Created a test suite using `unittest`  to validate the functionality of the module.
   
### Testing Instructions
1. Run `unittest` to execute the tests: `python -m unittest tests/feature_extraction_test.py`

### CODE REVIEW

File: fairseq-sl/fairseq/data/token_block_utils_fast.cpp [hunk 1]
```diff
@@ -1,28 +1,25 @@
...
 {
     "distutils": {
         "depends": [
-            "/tmp/pip-build-env-zoh18iqa/overlay/lib/python3.10/site-packages/numpy/core/include/numpy/arrayobject.h",
-            "/tmp/pip-build-env-zoh18iqa/overlay/lib/python3.10/site-packages/numpy/core/include/numpy/arrayscalars.h",
-            "/tmp/pip-build-env-zoh18iqa/overlay/lib/python3.10/site-packages/numpy/core/include/numpy/ndarrayobject.h",
-            "/tmp/pip-build-env-zoh18iqa/overlay/lib/python3.10/site-packages/numpy/core/include/numpy/ndarraytypes.h",
-            "/tmp/pip-build-env-zoh18iqa/overlay/lib/python3.10/site-packages/numpy/core/include/numpy/ufuncobject.h"
+            "C:\\Users\\myous\\AppData\\Local\\Temp\\pip-build-env-7b3b5wvo\\overlay\\Lib\\site-packages\\numpy\\_core\\include\\numpy\\arrayobject.h",
+            "C:\\Users\\myous\\AppData\\Local\\Temp\\pip-build-env-7b3b5wvo\\overlay\\Lib\\site-packages\\numpy\\_core\\include\\numpy\\arrayscalars.h",
```
Comment (antoine-tran) [inline fairseq-sl/fairseq/data/token_block_utils_fast.cpp:8]: I don't think changing a Unix-dependent (yet user-neutral) to a Windows-dependent (and user-specific) is a better refactoring choice

File: fairseq-sl/fairseq/data/data_utils_fast.cpp [hunk 2]
```diff
@@ -1,28 +1,25 @@
...
 {
     "distutils": {
         "depends": [
-            "/tmp/pip-build-env-zoh18iqa/overlay/lib/python3.10/site-packages/numpy/core/include/numpy/arrayobject.h",
-            "/tmp/pip-build-env-zoh18iqa/overlay/lib/python3.10/site-packages/numpy/core/include/numpy/arrayscalars.h",
-            "/tmp/pip-build-env-zoh18iqa/overlay/lib/python3.10/site-packages/numpy/core/include/numpy/ndarrayobject.h",
-            "/tmp/pip-build-env-zoh18iqa/overlay/lib/python3.10/site-packages/numpy/core/include/numpy/ndarraytypes.h",
-            "/tmp/pip-build-env-zoh18iqa/overlay/lib/python3.10/site-packages/numpy/core/include/numpy/ufuncobject.h"
+            "C:\\Users\\myous\\AppData\\Local\\Temp\\pip-build-env-7b3b5wvo\\overlay\\Lib\\site-packages\\numpy\\_core\\include\\numpy\\arrayobject.h",
+            "C:\\Users\\myous\\AppData\\Local\\Temp\\pip-build-env-7b3b5wvo\\overlay\\Lib\\site-packages\\numpy\\_core\\include\\numpy\\arrayscalars.h",
```
Comment (antoine-tran) [inline fairseq-sl/fairseq/data/data_utils_fast.cpp:8]: Same here. If you want we can use an OS environment instead

File: fairseq-sl/fairseq/data/token_block_utils_fast.cpp (hunk 1 above)
Comment (JooZef315) [reply fairseq-sl/fairseq/data/token_block_utils_fast.cpp]: retrieved them. Done

File: fairseq-sl/fairseq/data/data_utils_fast.cpp
Comment (heffernankevin) [inline fairseq-sl/fairseq/data/data_utils_fast.cpp:1]: Did you mean to add these `.cpp` files?

File: fairseq-sl/fairseq/data/token_block_utils_fast.cpp
Comment (heffernankevin) [inline fairseq-sl/fairseq/data/token_block_utils_fast.cpp:1]: Same for this one

File: fairseq-sl/pip
Comment (heffernankevin) [inline fairseq-sl/pip:1]: empty file / directory

File: tests/feature_extraction_test.py
Comment (heffernankevin) [inline tests/feature_extraction_test.py:1]: Nice idea of adding a `tests` directory! I would move the `test_env.py` you created earlier here

File: tests/feature_extraction_test.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import unittest
+from translation.feature_extraction_module import FeatureExtractionConfig, FeatureExtractionModule, LauncherConfig
+from unittest.mock import patch, MagicMock
+
+class TestFeatureExtractionModule(unittest.TestCase):
+    def setUp(self):
+        # Mock the configuration for the FeatureExtractionModule
+        self.config = FeatureExtractionConfig(
+            data_dir="MOCK_dataset",
+            pretrained_model_path="signhiera_mock.pth",
```
Comment (heffernankevin) [inline tests/feature_extraction_test.py:10]: Ideally, the test would also download this mock encoder for the user instead of expecting it to be downloaded in a specific place first

**Review by heffernankevin — CHANGES_REQUESTED** [review]

File: fairseq-sl/fairseq/data/data_utils_fast.cpp
Comment (JooZef315) [reply fairseq-sl/fairseq/data/data_utils_fast.cpp:1]: No, windows add them automatically.

Comment (heffernankevin) [reply fairseq-sl/fairseq/data/data_utils_fast.cpp:1]: Can you remove them from the PR? These shouldn't be added

File: utils/download_model.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+
+import os
+import wget
+
+def get_model_path():
```
Comment (heffernankevin) [inline utils/download_model.py:10]: Maybe we'd like to download other models in future. Can you refactor the URL out of this function so that is accepts that as input?

File: utils/download_model.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+
+import os
+import wget
+
+def get_model_path(url: str):
```
Comment (heffernankevin) [inline utils/download_model.py:10]: model_path: str = "signhiera_mock.pth"

File: README.md [hunk 6]
```diff
@@ -15,7 +15,7 @@ This repository contains research code for the paper [*Towards Privacy-Aware Sig
 
 
 
-SSVP-SLT relies on masked autoencoding (MAE) on anonymized videos as a form of self-supervised pretraining to learn continuous sign language representations at scale. The learned representations are transferred to the supervised gloss-free sign language translation task. SSVP-SLT outperforms prior SOTA methods on the ASL-to-English How2Sign benchmark in the finetuned and zero-shot settings by over 3 BLEU points. 
+SSVP-SLT relies on masked autoencoding (MAE) on anonymized and unannotated videos as a form of self-supervised pretraining to learn continuous sign language representations at scale. The learned representations are transferred to the supervised gloss-free sign language translation task. SSVP-SLT outperforms prior SOTA methods on the ASL-to-English How2Sign benchmark in the finetuned and zero-shot settings by over 3 BLEU points.
```
Comment (heffernankevin) [inline README.md:18]: Can you remove `and unannotated`? Let's leave the main README for now

File: tests/feature_extraction_test.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+
+import unittest
+from translation.feature_extraction_module import FeatureExtractionConfig, FeatureExtractionModule, LauncherConfig
```
Comment (heffernankevin) [inline tests/feature_extraction_test.py:8]: I think that this `tests` directory and the `translation` directory should go into the `src/ssvp_slt` so that you can import them normally e.g. `from ssvp_slt.translation.feature_extraction_module`

File: tests/feature_extraction_test.py [hunk 8]
```diff
@@ -0,0 +1,38 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```
Comment (heffernankevin) [inline tests/feature_extraction_test.py:1]: Let's rename to: `test_feature_extraction`

File: translation/feature_extraction_module.py [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+from einops import rearrange
+from typing import Any, Generator, List, Union, Tuple
+
+import pandas as pd
+import torch
+from torch import nn
+from torch.utils.data import DataLoader
+from tqdm import tqdm
+
+from src.ssvp_slt.data.video_dataset import VideoDataset
```
Comment (heffernankevin) [inline translation/feature_extraction_module.py:20]: You should remove `src` from all these imports.

File: utils/download_model.py [hunk 10]
```diff
@@ -0,0 +1,19 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```
Comment (heffernankevin) [inline utils/download_model.py:1]: Can you also move this into: `src/ssvp_slt/util/download_model.py`

File: README.md [hunk 11]
```diff
@@ -1,9 +1,8 @@
-<a href="https://pytorch.org/get-started/locally/"><img alt="PyTorch" src="https://img.shields.io/badge/PyTorch-ee4c2c?logo=pytorch&logoColor=white"></a>  [![report](https://img.shields.io/badge/ArXiv-Paper-blue)](https://arxiv.org/abs/2402.09611)
-
+<a href="https://pytorch.org/get-started/locally/"><img alt="PyTorch" src="https://img.shields.io/badge/PyTorch-ee4c2c?logo=pytorch&logoColor=white"></a> [![report](https://img.shields.io/badge/ArXiv-Paper-blue)](https://arxiv.org/abs/2402.09611)
```
Comment (heffernankevin) [inline README.md:1]: Can you revert the changes you made to this README?

Comment (JooZef315) [reply README.md]: done.

**Review by heffernankevin — APPROVED** [review]
